packages/default-theme/cms/elements/CmsElementProductListing.vue

```diff
@@ -1,56 +1,60 @@
 <template>
-  <div class="sw-product-list">
-    <SfLoader :loading="loading" class="sw-product-list__loader"/>
-      <div class="sw-product-list__wrapper" v-if="products.length">
-        <div class="sw-product-list__list" :class="{'sw-product-list__list--blur': loading}">
-          <SwProductCard
-            class="sw-product-list__product-card"
-            v-for="product in products"
-            :key="product.id"
-            :product="product"
-          />
-          <div class="sw-product-list__place-holder"/>
-        </div>
-        <SfPagination
-          class="sw-product-list__pagination desktop-only"
-          :current="pagination.currentPage"
-          :total="Math.ceil(pagination.total / pagination.perPage)"
-          :visible="5"
-          @click="changedPage"
+  <div class="cms-element-product-listing">
+    <SfLoader :loading="loading" class="cms-element-product-listing__loader" />
+    <div class="cms-element-product-listing__wrapper" v-if="products.length">
+      <div
+        class="cms-element-product-listing__list"
+        :class="{ 'cms-element-product-listing__list--blur': loading }"
+      >
+        <SwProductCard
+          class="cms-element-product-listing__product-card"
+          v-for="product in products"
+          :key="product.id"
+          :product="product"
         />
+        <div class="cms-element-product-listing__place-holder" />
       </div>
-      <SfHeading
-        v-else
-        title="No products found"
-        subtitle="let us look for them"
+      <SfPagination
+        class="cms-element-product-listing__pagination desktop-only"
+        :current="pagination.currentPage"
+        :total="Math.ceil(pagination.total / pagination.perPage)"
```

Comment:
would be nice to have that logic moved somewhere else I think :)

Reply:
sure, but with another PR :)